features/fixtures/mazerunner/src/main/java/com/bugsnag/android/mazerunner/scenarios/ManualSessionSmokeScenario.kt

```diff
@@ -0,0 +1,59 @@
+package com.bugsnag.android.mazerunner.scenarios
+
+import android.content.Context
+import com.bugsnag.android.Bugsnag
+import com.bugsnag.android.Configuration
+import android.os.Handler
+import android.os.HandlerThread
+import android.util.Log
+import com.bugsnag.android.flushAllSessions
+
+/**
+ * Sends an exception after pausing the session
+ */
+internal class ManualSessionSmokeScenario(config: Configuration,
+                                          context: Context) : Scenario(config, context) {
+
+    companion object {
+        private const val SLEEP_MS: Long = 300
+    }
+
+    init {
+        config.autoTrackSessions = false
+    }
+
+    override fun run() {
+        super.run()
+        val client = Bugsnag.getClient()
+        client.setUser("123", "ABC.CBA.CA", "ManualSessionSmokeScenario")
+        val thread = HandlerThread("HandlerThread")
+        thread.start()
+
+        Handler(thread.looper).post(Runnable {
```

Comment:
The sleeps are only necessary as we don't know when reports are delivered. It should be possible to get this information by implementing a custom delivery class, and then chaining requests one after the other. Something along these lines should do the trick:

```
val baseDelivery = config.delivery
config.delivery = InterceptingDelivery(baseDelivery)

class InterceptingDelivery(baseDelivery: Delivery, callback: () -> Unit): Delivery {
    override fun deliver() {
        baseDelivery.deliver()
        callback()
    } 
}
```